### ISSUE TITLE
[Feature request] move generated while make files into another directory

### ISSUE THREAD

**Issue by mdziczkowski** [pennmush/pennmush#1360]
I would like to request the feature that all files generated within the `make`, should be put into a separate folder outside the `src`. Additionally, the files, should be into sub-folders, basing on their extension. For example: *.o -> libs, binaries (netmush, etc.) -> bin

**Comment by HarryCordewener** [comment]
While I agree this is a good standard to have, what is the benefit to this feature request to the end user that will make it worth the time of a developer to do this work?

**Comment by mdziczkowski** [comment]
The `make` commands causes that all of the files (sources and results) are put together inside of the `src` folder, causing that it get a bit "messy" and the amount of the files makes that it's hard to find anything (for example: binaries) without doing a search.

There were cases that sometimes (from time to time, not regulatory) some binaries weren't generated (I had discovered that after the search and browsing the `src` folder.

Adding the proposed as an standard would keep the sources separated from the "results" (with is an advantage while the development and bug fixing)

**Comment by ray73864** [comment]
More than welcome to do a PR for it.

As for the 'discover if all files have become correctly generated', when you run 'make', it'll spit out an error if it couldn't compile a file and then stop compiling from that point onwards.

**Comment by mdziczkowski** [comment]
Not exactly. There were a cases that the make got finished correctly (without any serious errors), but the binaries (like for example "netmush") hasn't become generated. It's possible that could be a temporary problem with got fixed

**Comment by ray73864** [comment]
If the netmush binary in the src directory didn't get created, then the 'make' would have failed on the linking step right at the very end, and again would have said why it failed.

**Comment by mdziczkowski** [comment]
@ray73864  I try to do some attempt's to recreate it. If I success then I'll make a separate report about it. Let's hope it was only an temporary problem with got fixed ;-)

**Comment by talvo** [comment]
I don't see any harm in this, but I also don't care enough to do it myself :) If someone submitted a patch and no one else can see any reason it could cause issues I'd support.

**Comment by mdziczkowski** [comment]
I had done some research and had discovered that the netmush get's generated (in oposite to ssl_slave and the info_something ones), but I haven't discovered yet, why does the `make install` does not work correctly. I have done at the start the `./configure --prefix=somedir` before `make` and then `make install`. The most files (except the ones mentioned above in the branches) but they does not get installed to given by the `configure` place (I tryed even `make instal DESTDIR=somedir` and it didn't worked too).

I think that it would be a good idea to make that the generated executables were put (instead of beeing linked) into the `game` directory, that the game is ready to run after finishing the buid process. It's possible that the linking may (but don't have too) cause that after moving there can become broken links (unfortunatelly, the Linux developers haven't thought about coding it, to make that the path of the link source would be generated dynamically, that when moving the source file to nother location wouldn't break the links)

**Comment by mdziczkowski** [comment]
I had found (thanks RHostMush ;-) ] a code with could solve the discovery of (for example) database file names, with can become automatically updated in the config file. Given code is only the base to work at.

> INPUT_DB=${DATA_DIR}/`grep "^input" ${GAMENAME}.cnf | awk {printf("%s",substr($0,index($0,$2)))}'`
